src/ameba/rule/performance/first_last_after_filter.cr

```diff
@@ -0,0 +1,51 @@
+module Ameba::Rule::Performance
+  # This rule is used to identify usage of `first/last` calls that follow filters.
+  #
+  # For example, this is considered inefficient:
+  #
+  # ```
+  # [-1, 0, 1, 2].select { |e| e > 0 }.first
+  # [-1, 0, 1, 2].select { |e| e > 0 }.last
+  # ```
+  #
+  # And can be written as this:
+  #
+  # ```
+  # [-1, 0, 1, 2].find { |e| e > 0 }
+  # [-1, 0, 1, 2].reverse.find { |e| e > 0 }
```

Comment:
this reverse create new array, this should be much faster `[-1, 0, 1, 2].reverse_each.find { |e| e > 0 }`